docs/source/notebook-components/weaver_example.ipynb

```diff
@@ -1135,22 +1144,38 @@
     "\n",
     "print(\"Waiting for job completion with pooling monitoring of its status...\")\n",
     "\n",
-    "timeout = 20  # Define a timeout to abandon this monitoring. Process is relatively quick and shouldn't last too long.\n",
+    "# Define a timeout to abandon this monitoring. Process is relatively quick and shouldn't last too long.\n",
+    "# The process will be retried if failed to handle possible sporadic errors from the WPS remote provider.\n",
+    "# Stops on first maximum timout/retry reached, whichever happens first.\n",
+    "timeout = 60\n",
+    "retries = 10\n",
+    "attempt = retries\n",
     "delta = 5\n",
     "body = {}\n",
     "while timeout >= 0:\n",
     "    resp = requests.get(status_location, **WEAVER_TEST_REQUEST_XARGS)\n",
     "    assert resp.status_code == 200, \"Failed retrieving job status at location [{}]\".format(status_location)\n",
     "    body = resp.json()\n",
     "    timeout -= delta\n",
-    "    if body[\"status\"] in [\"accepted\", \"running\"]:\n",
-    "        print(f\"Delay: {delta}s, Duration: {body['duration']}, Status: {body['status']}\")\n",
+    "    status = body[\"status\"]\n",
+    "    if status in [\"accepted\", \"running\"]:\n",
+    "        print(f\"Delay: {delta}s, Duration: {body['duration']}, Status: {status}\")\n",
     "        time.sleep(delta)\n",
     "        continue\n",
-    "    if body[\"status\"] in [\"failed\", \"succeeded\"]:\n",
-    "        print(f\"Final job status: [{body['status']}]\")\n",
+    "    if status in [\"failed\", \"succeeded\"]:\n",
+    "        print(f\"Final job status: [{status}]\")\n",
+    "        if status == \"failed\":\n",
+    "            if attempt > 0:\n",
+    "                attempt -= 1\n",
+    "                retry_msg = f\"{retries - attempt}/{retries}\"\n",
+    "                print(f\"Retrying execution... ({retry_msg})\")\n",
+    "                status_location = submit_job()\n",
+    "                print(f\"Job Status Location: [{status_location}] (retry: {retry_msg})\")\n",
+    "                continue\n",
```

Comment:
So this loop will exit if one of timeout or attempt, first one to reach zero?  Just want to ensure no infinite loop.

Reply:
Yes. First one will break out.